grails-app/services/org/pih/warehouse/shipping/ShipmentService.groovy

```diff
@@ -1108,6 +1108,16 @@ class ShipmentService {
             // Add a Shipped event to the shipment
             createShipmentEvent(shipmentInstance, shipDate, EventCode.SHIPPED, locationInstance)
 
+            // Send notification email to recipients
+            def g = grailsApplication.mainContext.getBean('org.codehaus.groovy.grails.plugins.web.taglib.ApplicationTagLib')
+            def recipientItems = shipmentInstance.shipmentItems.groupBy {it.recipient }
+            recipientItems.each {recipient, items ->
+                def subject = g.message(code: "email.yourItemShipped.message", args: [shipmentInstance.shipmentNumber])
+                def body = "${g.render(template: "/email/shipmentItemShipped", model: [shipmentInstance: shipmentInstance, shipmentItems: items])}"
+                if (recipient) {
+                    mailService.sendHtmlMail(subject, body.toString(), recipient.email)
+                }
+            }
```

Comment:
Also check out line 1139. I was hoping (but did not explicitly say) that we'd implement the recipient email sending feature using the same event bus used to send the status change emails. If possible, we should move the email sending code above to its own method in the NotificationService and then invoke it using the ShipmentStatusTransitionEventService. If you have time please make that change. 

Otherwise if @awalkowiak is ok with the PR feel free to merge it so we can get Katarzyna to test before the sprint is closed.